README.md

```diff
@@ -1 +1,102 @@
 # Bauble
+
+Bauble is a data format targeted at games with rust-like syntax.
+
+It is intended to be used with a system that has some kind of reflection on types. So types can be optionally included instead of refering to them by full path. And it also gives the data-format the potential to load erased types.
+
+Bauble has three steps in parsing.
+1. text -> simple intermediate data representation. 
+2. -> intermediate data representation with resolved types.
+3. -> rust types through implementation of `FromBauble`. 
+
+## Examples of bauble
+```rust
+// bauble has the capability to use different types. Which are resolved by the `AssetCtx`
+use rpg::{Enemy, DamageType};
+
+slime = Enemy {
+  name: "Slime",
+  hp: 200,
+  speed: 2.5,
+  damage_type: DamageType::Slimy,
+  // You can link to assets by prefixing a path with `$`.
+  audio: $assets::sounds::slime_audio,
+  // Arrays, which could be deserialized as other containers, are deliminated by `[]`
+  attacks: [$slam_attack],
+}
+
+// Putting `copy` infront of a definition essentially makes it a macro i.e the value is 
+// copied in place instead of being a reference to it, which can be refered to using the
+// `$ident` syntax.
+//
+// As in rust, tuples are delimeted by `()`.
+copy slam_attack = (2.5, "Slam")
+```
+Bauble also supports raw pieces of data. This can be used for things like custom short-hand syntax (#2d6 dice) or even scripts. The user of the library defines how such raw values are parsed.
+```rust
+use rpg::{Ability, Projectile, OnUse};
+fireball = Ability {
+  name: "Fireball",
+  // You can link to other assets within the document.
+  on_use: OnUse::SpawnProjectile($fireball_projectile),
+}
+
+fireball_projectile = Projectile {
+  // After a `#` a raw value is accepted. This could be used for a color.
+  color: #42f56f,
+  // Or a dice roll.
+  hit_damage: #2d6,
+  // Everything within `#{ ... }` is considered raw data. Other instances of the character `{` within this raw data must be delimited by `}`.
+  //
+  // This can be used to have code within the format. 
+  on_hit: #{
+    ctx.spawn_effect("FireballEffect");
+    ctx.play_sound("FireballSound");
+    for enemy in ctx.nearby_enemies(1.0) {
+      enemy.damage(ctx.data.hit_damage);
+    }
+  }
```

Comment:
```suggestion
  on_hit: #{{
    ctx.spawn_effect("FireballEffect");
    ctx.play_sound("FireballSound");
    for enemy in ctx.nearby_enemies(1.0) {
      enemy.damage(ctx.data.hit_damage);
    }
  }}
```
Unless it was changed at some point, raw blocks that contain `}`s should use additional surrounding `{}`s to disambiguate. A note about this would be helpful too.

Reply:
This was necessary to avoid reimplementing Rust's delimiters, escapes, etc to make `bauble` support raw Rust blocks